Multiplication on closed intervals of rational numbers (#1351)
UniMath/agda-unimath: src/elementary-number-theory/maximum-rational-numbers.lagda.md

# Maximum on the rational numbers

```agda
module elementary-number-theory.maximum-rational-numbers where
```

<details><summary>Imports</summary>

```agda
open import elementary-number-theory.decidable-total-order-rational-numbers
open import elementary-number-theory.inequality-rational-numbers
open import elementary-number-theory.rational-numbers
open import elementary-number-theory.strict-inequality-rational-numbers

open import foundation.coproduct-types
open import foundation.identity-types
open import foundation.transport-along-identifications

open import order-theory.decidable-total-orders
```

</details>

## Idea

> We define the operation of maximum

Reviewer: ```suggestion
We define the operation of {{#concept "maximum" Disambiguation="of pairs of rational numbers" Agda=max-ℚ}}
```

([least upper bound](order-theory.least-upper-bounds-posets.md)) for the
[rational numbers](elementary-number-theory.rational-numbers.md).

## Definition

```agda
max-ℚ : ℚ → ℚ → ℚ
max-ℚ = max-Decidable-Total-Order ℚ-Decidable-Total-Order
```

## Properties

> ### Associativity of `max-ℚ`

Reviewer: ```suggestion
### Associativity of the maximum operation
```

```agda
associative-max-ℚ : (x y z : ℚ) → max-ℚ (max-ℚ x y) z ＝ max-ℚ x (max-ℚ y z)
associative-max-ℚ =
  associative-max-Decidable-Total-Order ℚ-Decidable-Total-Order
```

> ### Commutativity of `max-ℚ`

Reviewer: ```suggestion
### Commutativity of the maximum operation
```

```agda
commutative-max-ℚ : (x y : ℚ) → max-ℚ x y ＝ max-ℚ y x
commutative-max-ℚ =
  commutative-max-Decidable-Total-Order ℚ-Decidable-Total-Order
```

> ### `max-ℚ` is idempotent

Reviewer: ```suggestion
### The maximum operation is idempotent
```

# Maximum on the rational numbers

```agda
module elementary-number-theory.maximum-rational-numbers where
```

<details><summary>Imports</summary>

```agda
open import elementary-number-theory.decidable-total-order-rational-numbers
open import elementary-number-theory.inequality-rational-numbers
open import elementary-number-theory.rational-numbers
open import elementary-number-theory.strict-inequality-rational-numbers

open import foundation.coproduct-types
open import foundation.identity-types
open import foundation.transport-along-identifications

open import order-theory.decidable-total-orders
```

</details>

## Idea

> We define the operation of
> {{#concept "maximum" Disambiguation="of pairs of rational numbers" Agda=max-ℚ}}
> ([least upper bound](order-theory.least-upper-bounds-posets.md)) for the
> [rational numbers](elementary-number-theory.rational-numbers.md).

Reviewer: Perhaps this idea section could be structured a little better? E.g.

```suggestion
The {{#concept "maximum" Disambiguation="of pairs of rational numbers" Agda=max-ℚ}} of two [rational numbers](elementary-number-theory.rational-numbers.md) is the [greatest](elementary-number-theory.inequality-rational-numbers.md) rational number of the two. This is the [binary least upper bound](order-theory.least-upper-bounds-posets.md) in the [total order of rational numbers](elementary-number-theory.decidable-total-order-rational-numbers.md).
```
